<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
<title>LibSylph: Assertion Class Reference</title>
<link href="doxygen.css" rel="stylesheet" type="text/css">
</head><body>
<!-- Generated by Doxygen 1.5.8 -->
<div class="navigation" id="top">
  <div class="tabs">
    <ul>
      <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
      <li><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li class="current"><a href="classes.html"><span>Classes</span></a></li>
      <li><a href="files.html"><span>Files</span></a></li>
      <li><a href="pages.html"><span>Related&nbsp;Pages</span></a></li>
    </ul>
  </div>
  <div class="tabs">
    <ul>
      <li><a href="classes.html"><span>Class&nbsp;Index</span></a></li>
      <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
      <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
    </ul>
  </div>
</div>
<div class="contents">
<h1>Assertion Class Reference</h1><!-- doxytag: class="Assertion" --><!-- doxytag: inherits="Sylph::Exception" --><code>#include &lt;<a class="el" href="_debug_8h-source.html">Sylph/Core/Debug.h</a>&gt;</code>
<p>
<div class="dynheader">
Inheritance diagram for Assertion:</div>
<div class="dynsection">
<p><center><img src="class_assertion__inherit__graph.png" border="0" usemap="#_assertion__inherit__map" alt="Inheritance graph"></center>
<map name="_assertion__inherit__map">
<area shape="rect" id="node2" href="class_sylph_1_1_exception.html" title="This class provides a ready&#45;to&#45;use interface for exception handling." alt="" coords="5,5,139,35"></map>
<center><font size="2">[<a href="graph_legend.html">legend</a>]</font></center></div>

<p>
<a href="class_assertion-members.html">List of all members.</a><table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">virtual const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_assertion.html#0f84db344ea67d9084fa450af0e2261e">name</a> () const   throw ()</td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Returns the class name of this Exception.  <a href="#0f84db344ea67d9084fa450af0e2261e"></a><br></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">const char *&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_sylph_1_1_exception.html#3b56cc13b4004aa72e7ea86f034f968c">what</a> () const   throw ()</td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Returns the reason why this <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a> was thrown, i.e. the second parameter to <code>sthrow()</code>.  <a href="#3b56cc13b4004aa72e7ea86f034f968c"></a><br></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">void&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="class_sylph_1_1_exception.html#1038e3fb0e96d4733beb0ea03c4aad04">addTraceMessage</a> (const char *message) const   throw ()</td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">This function adds a message to the trace information inside this <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a>.  <a href="#1038e3fb0e96d4733beb0ea03c4aad04"></a><br></td></tr>
</table>
<hr><a name="_details"></a><h2>Detailed Description</h2>
<dl compact><dt><b><a class="el" href="todo.html#_todo000003">Todo:</a></b></dt><dd>Write documentation! </dd></dl>
<hr><h2>Member Function Documentation</h2>
<a class="anchor" name="1038e3fb0e96d4733beb0ea03c4aad04"></a><!-- doxytag: member="Assertion::addTraceMessage" ref="1038e3fb0e96d4733beb0ea03c4aad04" args="(const char *message) const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">void Sylph::Exception::addTraceMessage           </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>message</em>          </td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()<code> [inline, inherited]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
This function adds a message to the trace information inside this <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a>. 
<p>
It can be useful to add the name of a function the <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a> passes through (i.e. what <code>strace</code> does), or any other information you like to add to the <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a>. This information is purely diagnostic. <dl compact><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>message</em>&nbsp;</td><td>The message to add in the traceinfo log. </td></tr>
  </table>
</dl>

</div>
</div><p>
<a class="anchor" name="0f84db344ea67d9084fa450af0e2261e"></a><!-- doxytag: member="Assertion::name" ref="0f84db344ea67d9084fa450af0e2261e" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">virtual const char* Assertion::name           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()<code> [inline, virtual]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Returns the class name of this Exception. 
<p>
If you used the <code> S_CREATE_EXCEPTION(</code><em>name</em><code>)</code> macro, this will be automatically filled in for you. Otherwise you'll have to put in the name yourself. This is purely diagnostic information. <dl class="return" compact><dt><b>Returns:</b></dt><dd>The name of the class of this Exception. </dd></dl>

<p>Reimplemented from <a class="el" href="class_sylph_1_1_exception.html#16528d47441cf5ffb45bcdbd119d1661">Sylph::Exception</a>.</p>

</div>
</div><p>
<a class="anchor" name="3b56cc13b4004aa72e7ea86f034f968c"></a><!-- doxytag: member="Assertion::what" ref="3b56cc13b4004aa72e7ea86f034f968c" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">const char* Sylph::Exception::what           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td> const  throw ()<code> [inline, inherited]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Returns the reason why this <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a> was thrown, i.e. the second parameter to <code>sthrow()</code>. 
<p>
<dl class="return" compact><dt><b>Returns:</b></dt><dd>The reason why this <a class="el" href="class_sylph_1_1_exception.html" title="This class provides a ready-to-use interface for exception handling.">Exception</a> was thrown. </dd></dl>

</div>
</div><p>
</div>
<hr size="1" class="unhidden"><address style="text-align: right;"><small>
Documentation for LibSylph by Doxygen
</body>
</html>
